modules/Makefile.kubernetes

```diff
@@ -0,0 +1,153 @@
+  $(error KUBERNETES_NAMESPACE is undefined)
+endif
+
+export KUBERNETES_NAMESPACE
+
+#
+# Reference Docs:
+#  - https://cloud.google.com/container-engine/docs/kubectl/
+#
+## Display info about the kubernetes setup
+kubernetes\:info:
+	@echo -e "Cluster Namespace: $(call yellow,$(CLUSTER_NAMESPACE))"
+	@echo -e "Cluster Domain: $(call yellow,$(CLUSTER_DOMAIN))"
+	@echo -e "Kubernetes Namespace: $(call yellow,$(KUBERNETES_NAMESPACE))"
+	@echo -e "SSH Tunnel: $(call yellow,$(KUBECTL_SSH_TUNNEL))"
+	@echo -e "SSH User: $(call yellow,$(KUBECTL_SSH_USER))"
+
+## Create a new service
+kubernetes\:create-service:
+	$(call kubectl_create,service)
+
+## Create a new controller
+kubernetes\:create-controller:
+	$(call kubectl_create,controller)
+
+## Create a new app composed of a service and a controller
+kubernetes\:create:
+	@$(SELF) kubernetes:create-service kubernetes:create-controller
+
+## Replace an existing service with downtime
+kubernetes\:replace-service:
+	@echo -e "INFO: Replacing $(KUBERNETES_NAME) service on cluster $(call yellow,$(CLUSTER_NAMESPACE))..."
+	@$(SELF) kubernetes:delete-service kubernetes:create-service
+
+## Replace an existing controller with downtime
+kubernetes\:replace-controller:
+	@echo -e "INFO: Replacing $(KUBERNETES_NAME) controller on cluster $(call yellow,$(CLUSTER_NAMESPACE))..."
+	@$(SELF) kubernetes:delete-controller kubernetes:create-controller
+
+## Replace an app composed of a service and a controller with downtime
+kubernetes\:replace:
+	@$(SELF) kubernetes:replace-service kubernetes:replace-controller
+
+## Delete a new service
+kubernetes\:delete-service:
+	$(call kubectl_delete,service)
+
+## Delete a new controller
+kubernetes\:delete-controller:
+	$(call kubectl_delete,controller)
+
+## Delete a new app composed of a service and a controller
+kubernetes\:delete:
+	@$(SELF) kubernetes:delete-service kubernetes:delete-controller
+
+## Deploy controller and service; notify datadog
+kubernetes\:deploy:
+	$(NOTIFY_STARTING)
+	@$(SELF) kubernetes:deploy-controller kubernetes:deploy-service || $(NOTIFY_FAILURE)
+	$(NOTIFY_SUCCESS)
+
+## Update existing controller using rolling-update or create a new controller; do not notify datadog
+kubernetes\:deploy-controller:
+	$(eval CURRENT_CONTROLLER_NAME = $(shell $(KUBECTL_CMD) get rc --selector="app=$(KUBERNETES_NAME)" -o jsonpath="{.items[0].metadata.name}" 2>/dev/null))
+	@if [ -z "$(CURRENT_CONTROLLER_NAME)" ]; then \
+  	$(SELF) kubernetes\:create-controller; \
+  else \
+		echo -e "INFO: Performing Rolling Update on $(KUBERNETES_NAME) on cluster $(call yellow,$(CLUSTER_NAMESPACE))..."; \
+	  envsubst < "$(KUBERNETES_RESOURCE_PATH)/$(KUBERNETES_NAME)-controller.yml" | envsubst | \
+    	$(KUBECTL_CMD) rolling-update $(CURRENT_CONTROLLER_NAME) $(KUBECTL_SCHEMA_CACHE_DIR) -f - ; \
+	fi
+
+## Replace existing service causing downtime if serial of resource has changed or create a service if one does not already exist; do not notify datadog
```

Comment:
lets make this a private target to avoid showing up in make help and causing confusion